### PR TITLE
HF <> Paperswithcode dataset identifiers mapping

### PR DESCRIPTION
**Pull request by julien-c**
As discussed offline in February with the Paperswithcode team, and again in https://github.com/huggingface/huggingface_hub/pull/39#issuecomment-833641258, here's a quick'n'dirty mapping of our dataset identifiers to Paperswithcode's.

### The goal
The goal here is to let Paperswithcode parse our metadata (to that end we'll need to align the task names too) so that:
- HuggingFace hub metadata can be listed automatically on Paperswithcode
- we can link to leaderboards on Paperswithcode
- potentially Paperswithcode can also display on their dataset pages how to load the dataset with `datasets`

Out of 661 "canonical" (= not user-specific) datasets we currently have in [huggingface/datasets](https://github.com/huggingface/datasets):
- 146 have exact matches in Paperswithcode
- 123 can be trivially mapped (only string differences are `[-_]`) to Paperswithcode
- 392 can't be trivially mapped.

Mapping code implmenting this simple `[-_]-independent` heuristic can be found in this [gist](https://gist.github.com/julien-c/e6eecd4afefbe264b424e02a23307c08)

### TODO for contributors 🥺
**Contributors, can you please** sample a few of those dataset ids to check that:
- no dataset is incorrectly mapped (unlikely)
- datasets listed in `unknown.csv` do not indeed have a corresponding dataset in Paperswithcode

If you find some that are not easily resolvable via simple heuristics, please create a `manual-mapping.yml` that we'll have to maintain over time.

### CODE REVIEW

File: docs/paperswithcode/datasets/manual-mapping.yml [hunk 1]
```diff
@@ -4,3 +4,42 @@ wmt16: wmt-2016
...
+poem_sentiment: gutenberg-poem-dataset
+omp: one-million-posts-corpus
+newsph: newsph-nli
+neural_code_search: neural-code-search-evaluation-dataset
+ncbi_disease: ncbi-disease-1
+multi_x_science_sum: multi-xscience
+multi_woz_v22: multiwoz
+multi_nli_mismatch: multinli
+mrqa: mrqa-2019
+meta_woz: metalwoz
```
Comment (julien-c) [inline docs/paperswithcode/datasets/manual-mapping.yml:45]: wow there's a lot!

**Comment by lhoestq** [comment]
I just finished mapping the datasets that we had in the unknown.csv :)

**Comment by julien-c** [comment]
Awesome! I think this is ready to merge then. We should try to keep it up to date while we add new "canonical" datasets to `datasets`. What would be the best process to do this?

It could also be an (optional) field in the dataset card, now that I think about it... Thoughts?

**Comment by julien-c** [comment]
## Heading over to https://github.com/huggingface/datasets/pull/2404 🎉
Will close this as we'll just store those as `paperswithcode_id` in dataset cards themselves.

**Comment by julien-c** [comment]
live mapping can be found at https://huggingface.co/api/pwc/datasets-mapping and will be kept up to date going forward

**Pull request closed without merging**